libgc/os_dep.c

```diff
@@ -26,7 +26,7 @@
 #     define __KERNEL__
 #     include <asm/signal.h>
 #     undef __KERNEL__
-#   else
+#   elif defined(__GLIBC__)
```

Comment:
If you look a few lines down there are already `__GLIB__` defines, the purpose is to skip those completely as logic like `#if 2 <= __GLIBC__` doesn't work if the symbol isn't defined at all.